### PR TITLE
feat(autofix): Add support for streamed output

### PR DESCRIPTION
**Pull request by roaga**
1. Adds a placeholder card that renders streamed text output
2. Updates animations to fit better
3. Reduces polling interval to 500 ms.

Screenshot 2024-12-12 at 9 31 36 AM

### CODE REVIEW

**Review by trillville — APPROVED** [review]

File: static/app/components/events/autofix/autofixOutputStream.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+export function AutofixOutputStream({step}: Props) {
+  const [displayedText, setDisplayedText] = useState('');
+  const intervalRef = useRef<number>();
+  const previousText = useRef('');
+  const currentIndexRef = useRef(0);
+
+  useEffect(() => {
+    if (!step.output_stream) {
+      return undefined;
```
Comment (jennmueng) [inline static/app/components/events/autofix/autofixOutputStream.tsx:32]:
```suggestion
      return;
```
Return by itself should already return `undefined`

File: static/app/components/events/autofix/autofixOutputStream.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        window.clearInterval(intervalRef.current);
+      }
+    }, 15);
+
+    return () => {
+      if (intervalRef.current !== undefined) {
+        window.clearInterval(intervalRef.current);
+      }
+    };
+  }, [displayedText, step.output_stream]);
```
Comment (jennmueng) [inline static/app/components/events/autofix/autofixOutputStream.tsx:58]: Is the `intervalRef` even needed in this case if it's not used outside the `useEffect`? Let's do
```
const interval = window.setInterval(...
```
and we can have a more sure clearing step on unmount with
```
window.clearInterval(interval)
```

File: static/app/components/events/autofix/autofixOutputStream.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+  if (!step.output_stream) {
```
Comment (jennmueng) [inline static/app/components/events/autofix/autofixOutputStream.tsx:60]: We're only using `output_stream` from the step right? Why not just pass only the output stream as a single prop, this way we can just not render this component at all if it doesn't exist

File: static/app/components/events/autofix/autofixOutputStream.tsx (hunk 1 above)
Comment (jennmueng) [reply static/app/components/events/autofix/autofixOutputStream.tsx:32]: See following comments, might not need this at all,

File: static/app/components/events/autofix/autofixSteps.tsx [hunk 4]
```diff
@@ -280,6 +281,7 @@ export function AutofixSteps({data, groupId, runId}: AutofixStepsProps) {
             </div>
           );
         })}
+        <AutofixOutputStream step={lastStep} />
```
Comment (jennmueng) [inline static/app/components/events/autofix/autofixSteps.tsx:284]: see other comment, don't even need to render this if the step doesn't have a stream

**Review by jennmueng — APPROVED** [review]
lgtm

**Pull request merged**